packages/backend-core/src/domain/comments/countDenormalization.ts

```diff
@@ -16,23 +16,35 @@ limitations under the License.
 
 import {
   CommentFlag,
-  CommentRecommendation,
   ICommentInstance,
 } from '../../models';
 
 export async function denormalizeCountsForComment(comment: ICommentInstance) {
-  const [
-    flaggedCount,
-    recommendedCount,
-  ] = await Promise.all([
-    CommentFlag.count({ where: { commentId: comment.id } }),
-    CommentRecommendation.count({ where: { commentId: comment.id } }),
-  ]);
-
-  const updatedComment = await comment.update({
-    flaggedCount,
-    recommendedCount,
-  });
+  let unresolvedFlagsCount = 0;
+  const flagsSummary: {[key: string]: Array<number>} = {};
+
+  const flags = await CommentFlag.findAll({ where: { commentId: comment.id } });
+
+  for (const f of flags) {
+    const flag = f.get();
+
+    if (!flagsSummary[flag.label]) {
+      flagsSummary[flag.label] = [0, 0, 0];
+    }
 
-  return Promise.resolve(updatedComment);
+    flagsSummary[flag.label][0] += 1;
```

Comment:
Seems to me like there is a semantics to the offset number 0 = total, [1] is resolved, [2] is recommended, in which case it would be better style to define constants that describe the meaning of the offset, and use those?